airbyte-server/src/main/java/io/airbyte/server/ServerApp.java

```diff
@@ -279,6 +293,13 @@ public static ServerRunnable getServer(final ServerFactory apiFactory, final Con
     }
   }
 
+  @VisibleForTesting
+  static boolean isLegalUpgrade(final AirbyteVersion airbyteDatabaseVersion, final AirbyteVersion airbyteVersion) {
```

Comment:
Nit:

the type could be Optional<AirbyteVersion> for the airbyteDatabaseVersion and the return can be:

```java
return !airbyteDatabaseVersion.map(version -> version.lessThan(VERSION_BREAK)
        && airbyteVersion.greaterThan(VERSION_BREAK))
.orElse(false);
```

about the version object do we have lessOrEqualThan functions so we can remove the first not. The double negation on the null check was confusing.

Reply:
```
return airbyteDatabaseVersion.map(version -> version.greatOrEqualThan(VERSION_BREAK)
        || airbyteVersion.lessOrEqualThan(VERSION_BREAK))
.orElse(true);
```